plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -183,13 +200,21 @@ static String getParametersString(Map<String, String> parametersMap) {
     return resultBuilder.toString();
   }
 
-  public static void showOptInDialog() {
+  /**
+   * @param parentShell if null, shows the dialog at the workbench level.
+   */
+  public static void showOptInDialog(Shell parentShell) {
     if (!hasUserOptedIn() && !hasUserRegisteredOptInStatus()) {
-      IWorkbenchWindow window = PlatformUI.getWorkbench().getActiveWorkbenchWindow();
-      if (window != null) {
-        new OptInDialog(window.getShell()).open();
+      if (parentShell != null) {
```

Comment:
I would move this logic into `findShell(parentShell)`.  `findShell(parentShell)` would:
  - return `parentShell` if not null
  - look for the active workbench window and return its window
  - look for `Display.getCurrent().getActiveShell()` providing there's a Display
  - return null

`null` is a valid parent shell: it just means the dialog is un-parented.